File: api/Core/Domain/Entities/Lobby.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Entities;

[Table("Lobbies")]
public class Lobby
{
    //TODO: Add limit to users and connections in lobby
    [Key]
    public int Id { get; set; }
    public string LobbyName { get; set; }
    public AppUser LobbyCreator { get; set; }
    public Game CurrentGame { get; set; }
    public ICollection<Game> PreviousGames { get; set; } = new List<Game>();
    public ICollection<AppUser> Users { get; set; }
    public ICollection<Connection> Connections { get; set; }
}

File: client/src/app/_models/lobby.ts
import { Game } from "./game";
import { GameMode } from "./gameMode";
import { LobbyUser } from "./user";

export interface Lobby {
    lobbyName: string;
    lobbyCreator: LobbyUser;
    users: LobbyUser[];
    pendingConnections: string[];
    gameMode: GameMode;
    currentGame?: Game;
    previousGames?: Game[];
}